### ISSUE TITLE
Sweep_: format wandb run name in pytorch checkpoint

### ISSUE THREAD

**Issue by ayulockin** [ayulockin/pytorch-lightning#1]
### Description & Motivation

It would be great to format the wandb run name: 
`wandb.run.name`

into the checkpoint filename.

epoch: 10
loss: 0.5423

filename: "epoch{epoch}-loss{val_loss:.2f}" --> "epoch10-loss0.54"

filename: "{wandb.run.name}-{epoch}-{val_loss:.2f}" --> "swept_sky_54-epoch10-loss0.54"

I am not sure if this is already possible but I think the filename function in the ModelCallback can only access metrics so far. 
 Suggested Solution: 
 ### Solution

PyTorch Lightning's current mechanism for formatting the filename in the checkpoint does indeed limit the scope of data that can be used for formatting. It primarily allows access to the trainer.properties and metrics.

In order for the `wandb.run.name` to be included in the formatting of the checkpoint filename, it would have to be made available within the context in which the filename formatting happens. 

One possible approach to address this issue might be to enhance the Checkpoint Callback function to also accept a dict of `extra_format_keys` which would then be made available in the filename formatting context. Accordingly, this dict could be updated with any additional information, like the `wandb.run.name`, before triggering the checkpoint.

```python
extra_format_keys = {
    'wandb_run_name': wandb.run.name
}
model_checkpoint = ModelCheckpoint(filename="{wandb_run_name}-{epoch}-{val_loss:.2f}", 
                                   extra_format_keys=extra_format_keys)

# then, in trainer
trainer = pl.Trainer(callbacks=[model_checkpoint])
```

We would then modify the `ModelCheckpoint._format_checkpoint_name` function to take into consideration the `extra_format_keys` while formatting the filename.

```python
def _format_checkpoint_name(self, filename, epoch, step, metrics, extra_format_keys={}):
    # default filename
    if filename is None:
        # logs_dir / version / checkpoints / 'epoch=5.ckpt'
        filename = "{epoch}" + self.CHECKPOINT_JOIN_CHAR + "{step}"

    # check and parse user passed keys in the string
    groups = re.findall(r'(\{.*?)[:\}]', filename)
    for tmp in groups:
        name = tmp[1:].split(":")[0]
        name = name.rstrip('}')  # if used in combination with typing e.g. {loss:.2f}
        if name not in metrics and name not in extra_format_keys and name not in self.CHECKPOINT_JOIN_CHAR:
            log.warning(
                f"Checkpoint filename pattern '{filename}' has '{tmp}' but it is not available in "
                "output dict. HINT: metrics are tensors and require to be cast to a value with "
                "`.item(), .as_tensor() or .tolist()`."
            )
        else:
            metrics[name] = metrics[name].item() if torch.is_tensor(metrics[name]) else metrics[name]

    # add all strings around
    groups = re.split(r'\{.*?\}', filename)
    callbacks = [w for w in zip(groups, list(metrics.keys()) + [""] * (len(groups) - len(metrics)))]
    out = ''.join([c + str(metrics.get(n, "")) for c, n in callbacks])
    return out
```
Here, the extra_format_keys dictionary is added as an argument to the `_format_checkpoint_name` function and the keys and values from the extra_format_keys are used to format filename.

While the above solution should work, it's crucial that we test it thoroughly to ensure that it doesn't break existing functionalities and works as expected.

Please note, this requires changing the PyTorch Lightning code so you might want to make a PR for this feature request if you want it included in the main PyTorch Lightning framework. However, before making a PR, make sure the team accepts this feature otherwise it might not get merged.

Additionally, implications, edge cases, or potential conflicts of adding `extra_format_keys` need to be thoroughly analyzed and tested. It's also worth to bring this issue up with the wider PyTorch Lightning community to get feedback and thoughts on how to best approach this.
